### PR TITLE
Add unittest runner

### PR DESCRIPTION
**Pull request by kazk**
```d
module solution;

export int add(int a, int b) {
    import std.stdio : stdout;
    stdout.writefln("a = %d, b = %d", a, b);
    return a;
}
```
```d
module solution_test;

import solution : add;

version(unittest) import fluent.asserts;

@("failing add with fluent asserts")
unittest {
    add(1, 1).should.equal(2).because("1 + 1 == 2");
}

@("failing add with assert")
unittest {
    assert(add(1, 1) == 2);
}
```

image

### CODE REVIEW

File: unittested.d [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+void onPassed(Test test, TestResult result) {
+    stdout.writefln("\n<PASSED::>Test Passed");
+}
+
+void onFailed(Test test, TestResult result) {
+    import std.algorithm : canFind;
+
+    foreach (th; result.thrown) {
+        if (th.type == "core.exception.AssertError" || th.type == "fluentasserts.core.base.TestException") {
```
Comment (kazk) [inline unittested.d:72]: Is there a better way to detect assertion failures?

**Comment by Imperatorn** [comment]
Well done ☀️. I'm on vacation with the family atm tho so can't review, sorry.

@maxhaton what do you think?

File: unittested.d (hunk 1 above)
Comment (maxhaton) [reply unittested.d:72]: Why are you saving the type and not the actual exception object?

Comment (kazk) [reply unittested.d:72]: This was derived from `silly`, and that's what they had (https://gitlab.com/AntonMeep/silly/-/blob/master/silly.d#L265).

If I keep the actual exception, is the syntax to check the type `if (cast(AssertError) exception)`?
For `fluentasserts.core.base.TestException`, don't I need that type resolvable? I want to keep fluent asserts optional.

**Pull request merged**